sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,199 @@ class TableCheckpointStore:
+                  the new checkpoint will be associated with.
+        :rtype: None
+        """
+        checkpoint_entity = self._create_checkpoint_entity(checkpoint)
+        try:
+            self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=checkpoint_entity)
+        except ResourceNotFoundError:
+            self.table_client.create_entity(entity=checkpoint_entity)
 
-    def update_checkpoint(self, checkpoint, **kwargs):
-        pass
+    def _update_ownership(self, ownership):
+        ownership_entity = self._create_ownership_entity(ownership)
+        try:
+            if ownership['etag'] is None:
+                metadata = self.table_client.create_entity(entity=ownership_entity)
+                ownership['etag'] = metadata['etag']
+                ownership['last_modified_time'] = metadata['date']
+                return ownership
+            metadata = self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=ownership_entity,
+            etag=ownership['etag'], match_condition=MatchConditions.IfNotModified)
+            ownership['etag'] = metadata['etag']
+            ownership['last_modified_time'] = metadata['date']
+            return ownership
+        except (ResourceModifiedError, ResourceExistsError):
+            raise OwnershipLostError()
 
-    def claim_ownership(self, ownershiplist, **kwargs):
-        pass
+    def claim_ownership(self, ownershiplist):
+        # type: (Iterable[Dict[str, Any]], Any) -> Iterable[Dict[str, Any]]
+        """Tries to claim ownership for a list of specified partitions.
+        :param Iterable[Dict[str,Any]] ownership_list: Iterable of dictionaries containing all the ownerships to claim.
+        :rtype: Iterable[Dict[str,Any]], Iterable of dictionaries containing partition ownership information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the ownership are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `owner_id` (str): A UUID representing the owner attempting to claim this partition.
+                - `last_modified_time` (UTC datetime.datetime): The last time this ownership was claimed.
+                - `etag` (str): The Etag value for the last time this ownership was modified. Optional depending
+                  on storage implementation.
+        """
+        newlist = []
+        for x in ownershiplist:
+            newownership = x
+            newlist.append(self._update_ownership(newownership))
```

Comment:
@Jg1255 , I think we still want the `_claim_one_partition` method, which catches/deals with different exceptions with logging, here instead of going straight to `_update_ownership`. Otherwise, we would change/lose the behavior of this method when those exceptions occur.
I think I was unclear when I mentioned that there was a bug, sorry about that :P 

To be more clear, [here is a draft PR](https://github.com/Azure/azure-sdk-for-python/pull/20008) with my changes of how this would look. (I still need to test these changes, just haven't had a chance to do that yet.) The biggest thing to note here is that we create a deep copy of the `ownership` dict in `_claim_one_partition` and pass that around, rather than modifying the original.

Reply:
ok I am going to fix and commit your suggests to the files